services/rag_service/embedding_service.py

```diff
@@ -0,0 +1,125 @@
+from services.data_service import DataService, get_data_service
+from services.github_service import GithubService
+from services.clients.anyscale_client import AnyscaleClient
+from services.clients.pinecone_client import PineconeClient
+from services.rag_service.text_chunker import TextChunker
+from fastapi import HTTPException, status
+from schemas.documentation_generation import FirestoreDoc, FirestoreRepo,EmbeddingModelEnum, StatusEnum
+from collections import deque 
+from routers import utils
+from dotenv import load_dotenv
+import firebase_admin
+import os
+import asyncio
+
+
+class EmbeddingService:
+    def __init__(
+        self, 
+        embedding_client: AnyscaleClient, 
+        vector_database_client: PineconeClient, 
+        github_service: GithubService, 
+        data_service: DataService, 
+        text_chunker: TextChunker
+    ):
+        self.embedding_client = embedding_client
+        self.vector_database_client = vector_database_client
+        self.github_service = github_service
+        self.data_service = data_service
+        self.text_chunker = text_chunker
+
+    async def generate_markdown_embeddings_for_repo(self, repo_id: str, user_id: str):
+        namespaces = vector_database_client.describe()["namespaces"]
```

Comment:
I'm going to backtrack a little here. The limitations for namespaces using Pinecone serverless are much bigger and better than I thought, so I'm more than happy to use them then:
https://docs.pinecone.io/docs/limits#serverless-index-limitations

I'll leave the comment above anyways, in case it provides any value.

Reply:
They don't even allow metadata filtering right now. We have no choice. However, like you said, they give us more than enough namespaces

<img width="727" alt="image" src="https://github.com/KTujjar/rocketdocs/assets/44108244/79102406-bb4f-4397-817a-a4af319b100e">